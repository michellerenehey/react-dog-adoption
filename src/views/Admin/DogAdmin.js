import { useState } from 'react';
import { addDog } from '../../services/dogs';
import { useHistory } from 'react-router-dom';
import DogForm from '../../components/Dog/DogForm';

export default function DogAdmin() {
  const [name, setName] = useState('');
  const [breed, setBreed] = useState('');
  const [image, setImage] = useState('');
  const [bio, setBio] = useState('');

  const history = useHistory();

  const handleSubmit = async (e) => {
    e.preventDefault();
    const data = await addDog(name, breed, image, bio);
    history.push(`/dogs/${data[0].id}`);
  };
  return (
    <div>
      <DogForm
        name={name}
        setName={setName}
        breed={breed}
        setBreed={setBreed}
        image={image}
        setImage={setImage}
        bio={bio}
        setBio={setBio}
        handleSubmit={handleSubmit}
      />
    </div>
  );
}
