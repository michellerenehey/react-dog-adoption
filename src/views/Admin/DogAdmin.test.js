import { render, screen } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import DogAdmin from './DogAdmin';

test('should render the dog form with admin access', async () => {
  const { container } = render(
    <MemoryRouter>
      <DogAdmin />
    </MemoryRouter>
  );
  await screen.findByText('Dog Name:');
  expect(container).toMatchSnapshot();
});
