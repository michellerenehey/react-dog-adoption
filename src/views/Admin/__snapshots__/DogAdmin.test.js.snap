// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render the dog form with admin access 1`] = `
<div>
  <div>
    <p
      class="message"
    />
    <div
      class="dog-form-div"
    >
      <form
        class="dog-form"
      >
        <label
          class="form-detail"
        >
          Dog Name:
        </label>
        <input
          class="form-detail"
          placeholder="enter name"
          required=""
          type="text"
          value=""
        />
        <label>
          Dog Breed:
        </label>
        <input
          placeholder="enter breed"
          type="text"
          value=""
        />
        <label>
          Image Link:
        </label>
        <input
          placeholder="enter image url"
          type="text"
          value=""
        />
        <label>
          Dog Bio:
        </label>
        <textarea
          class="dog-bio"
          placeholder="enter bio"
        />
        <button>
          Save
        </button>
      </form>
    </div>
  </div>
</div>
`;
